pkg/tools/topo-generator/parser/convert.go

```diff
@@ -0,0 +1,152 @@
+package parser
+
+import (
+	"strings"
+	"regexp"
+	"strconv"
+	"github.com/google/uuid"
+	"github.com/onosproject/onos-topo/pkg/tools/topo-generator/reader"
+)
+
+type Underlay struct {
+	Networks []Network 
+}
+
+type Network struct {
+    Entity_Id       string      
+	Name			string		
+    Display_Name    string  
+	Switches        []Switches 
+	Links			[]Link		
+}
+
+type Switches struct {
+    Entity_Id		string          
+	Name			string
+	Model_Id		string	        
+	Role			string	        
+	P4RT_Address	string
+	P4RT_Port		int
+	Insecure		bool
+	Ports           []Ports 
+}
+
+type Ports struct {
+	Entity_Id		string	
+	Name			string
+	Display_Name	string	
+	Speed			string	
+	Port_Number		int		
+	Channel_Number	int		
+}
+
+type Link struct {
+	Source			string	
+	Source_Name		string
+	Destination		string	
+	Dest_Name		string
+	Link_Type		string
+	// for both unidirectional and bidirectional
+	URI					string
+	URI_Name			string
+	UUID_Source			string
+	UUID_Dest			string
+	UUID_Source_Name	string
+	UUID_Dest_Name		string
+	// for bi-directional case (default)
+	URI1 				string
```

Comment:
Why we have two sets of URI, URI_NAME etc?

Reply:
This is for the bidirectional case. If you look at bidirectional.yaml in the templates folder it makes more sense. 
It is so that we can have both, where we make sure that the URI is flipped for the other set of originates and terminates. Since it is a struct, I needed to create separate variables in the struct for that.
p4rt:2/1/0-p4rt:1/1/0        link            <None>   <None>
p4rt:1/1/0-p4rt:2/1/0        link            <None>   <None>

Reply:
I have changed the names now.